### PR TITLE
Plfm 2390 b

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	private Long startedOn;
+	
+	@Field(name = COL_ASYNCH_JOB_STARTED_BY, nullable = false)
+	@ForeignKey(table = SqlConstants.TABLE_USER_GROUP, field = SqlConstants.COL_USER_GROUP_ID, cascadeDelete = true)
+	private Long startedByUserId;
+	
+	@Field(name = COL_ASYNCH_JOB_CHANGED_ON, nullable = false)
+	private Long changedOn;
+	
+	@Field(name = COL_ASYNCH_JOB_COMPRESSED_BODY, blob="mediumblob", nullable = false)
```
Comment (marcel-blonk) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:93]: serialized="mediumblob" will take care of serializing/deserializing for you

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+	public String getErrorMessage() {
+		return errorMessage;
+	}
+
+	public void setErrorMessage(String errorMessage) {
+		this.errorMessage = errorMessage;
+	}
+
+	public byte[] getErrorDetails() {
```
Comment (marcel-blonk) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:147]: why byte[] and not string?

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	private Long progressCurrent;
+
+	@Field(name = COL_ASYNCH_JOB_PROGRESS_TOTAL, nullable = true)
+	private Long progressTotal;
+	
+	@Field(name = COL_ASYNCH_JOB_PROGRESS_MESSAGE, nullable = true, varchar=MAX_MESSAGE_CHARS)
+	private String progressMessage;
+	
+	@Field(name = COL_ASYNCH_JOB_STARTED_ON, nullable = false)
+	private Long startedOn;
```
Comment (marcel-blonk) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:84]: Date?

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/AsynchJobType.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	}
+	
+	/**
+	 * Lookup the Type for a given class
+	 * @param clazz
+	 * @return
+	 */
+	public static AsynchJobType findType(Class<? extends AsynchronousJobBody> clazz){
+		for(AsynchJobType type: AsynchJobType.values()){
+			if(type.clazz.equals(clazz)) return type;
```
Comment (marcel-blonk) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/AsynchJobType.java:30]: insert new line, hard to scan code

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	
+	/**
+	 * The maximum number of characters in a string message.
+	 */
+	public static final int MAX_MESSAGE_CHARS = 3000;
+	
+	/**
+	 * State of a job
+	 */
+	public enum JobState{
```
Comment (marcel-blonk) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:48]: Use the one from autogenerated?

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/asynch/AsynchronousJobStatus.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+			"description": "The job's state can be one of the following enumerations",
+			"name": "AsynchJobState",
+			"id": "org.sagebionetworks.repo.model.asynch.AsynchJobState",
+			"enum": [
+				"PROCESSING",
+				"FAILED",
+				"COMPLETE"
+			]
+		},
+		"jobBody":{
```
Comment (marcel-blonk) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/asynch/AsynchronousJobStatus.json:15]: Maybe split in request and response body?

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java [hunk 7]
```diff
@@ -245,4 +247,23 @@ public boolean isAnonymousUser(UserInfo userInfo) {
 		if(userInfo == null) throw new IllegalArgumentException("UserInfo cannot be null");
 		return AuthorizationUtils.isUserAnonymous(userInfo);
 	}
+
+	@Override
+	public boolean canUserStartJob(UserInfo userInfo, AsynchronousJobBody bodyIntf) throws DatastoreException, NotFoundException {
+		if(bodyIntf == null) throw new IllegalArgumentException("Body cannot be null");
+		// Anonymous cannot start a job
+		if(AuthorizationUtils.isUserAnonymous(userInfo)) return false;
```
Comment (marcel-blonk) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java:255]: new line?

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/asynch/AsynchJobStatusManagerImpl.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	AuthorizationManager authorizationManager;
+	@Autowired
+	StackStatusDao stackStatusDao;
+
+	
+	@Override
+	public AsynchronousJobStatus getJobStatus(UserInfo userInfo, String jobId) throws DatastoreException, NotFoundException {
+		if(userInfo == null) throw new IllegalArgumentException("UserInfo cannot be null");
+		// Get the status
+		AsynchronousJobStatus status = asynchJobStatusDao.getJobStatus(jobId);
```
Comment (marcel-blonk) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/asynch/AsynchJobStatusManagerImpl.java:32]: throw if stack readonly and in process

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/AsynchronousJobController.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+	 * Once a job is launched its progress can be monitored by getting its status with this method.
+	 * @param userId
+	 * @param jobId The jobId issued to a job that has been launched with <a href="${POST.asynchronous.job}">POST /asynchronous/job</a>
+	 * @return
+	 * @throws NotFoundException 
+	 */
+	@ResponseStatus(HttpStatus.OK)
+	@RequestMapping(value = UrlHelpers.ASYNCHRONOUS_JOB_ID, method = RequestMethod.GET)
+	public @ResponseBody
+	AsynchronousJobStatus launchNewJob(
```
Comment (marcel-blonk) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/AsynchronousJobController.java:72]: name change

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java (hunk 1 above)
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:93]: For this case the serialization alias changes at runtime so cannot use an annotation for the alias.

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java (hunk 5 above)
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/asynch/DBOAsynchJobStatus.java:48]: Convinced myself a layer of abstraction from the API to DB is worth it.

**Pull request merged**